src/app/components/Application/Application.jsx

```diff
@@ -71,6 +38,36 @@ class App extends React.Component {
 
   componentDidMount() {
     Store.listen(this.onChange);
+    // Listen to the browser's navigation buttons.
+    this.props.route.history.listen((location = { action: {}, search: {}, query: {} }) => {
```

Comment:
I think `action` and `search` always a string? The default value probably should be the same data type?

Reply:
Ah ok, I've changed them to empty strings